packages/compiler/test/shadow_css_spec.ts

```diff
@@ -195,6 +195,12 @@ export function main() {
         expect(s(':host-context([a=b]) {}', 'contenta', 'a-host'))
             .toEqual('[a=b][a-host], [a="b"] [a-host] {}');
       });
+
+      it('should handle multiple host-context', () => {
+        expect(s(':host-context(.one) :host-context(.two) {}', 'contenta', 'a-host'))
+            .toEqual(
+                '.one[a-host] .two[contenta], .one .two [a-host], .two[a-host] .one[contenta], .two .one [a-host], .one.two[a-host], .one.two [a-host] {}');
+      });
```

Comment:
Just rebasing and tidying up this PR. I think this test is not actually correct. 

First, I don't think that there should be a space between the `:host-context()` selectors in the string passed to `s()`. Having `:host-context(...) :host-context(...)` doesn't make sense to me. I am not even sure how we should interpret this.

Second, I think that `contenta` should not appear in the final selection at all.

This is what I would expect:

```suggestion
expect(s(':host-context(.one):host-context(.two) {}', 'contenta', 'a-host')
            .replace(/ \{\}$/, '')
            .split(/\,\s+/)
            .sort())
    .toEqual([
      '.one .two[a-host]',   // `one` is an ancestor and `two` is on the host
      '.one .two [a-host]',  // `one` and `two` are both ancestors (in that order)
      '.one.two [a-host]',   // `one` and `two` are both on the same ancestor
      '.one.two[a-host]',    // `one` and `two` both on the host
      '.two .one [a-host]',  // `two` and `one` are both ancestors (in that order)
      '.two .one[a-host]',   // `two` is an ancestor and `one` is on the host
    ]);
```